pkg/controllers/management/agentupgrade/controller.go

```diff
@@ -40,20 +44,24 @@ func Register(ctx context.Context, context *config.ManagementContext) {
 	context.Apps.DaemonSets("").Controller().AddHandler(ctx, "agent-upgrade", h.OnDaemonSetChange)
 }
 
+// OnDeploymentChange checks if it should delete a namespaced deployment and if so, deletes it.
 func (h *handler) OnDeploymentChange(key string, deploy *appsv1.Deployment) (runtime.Object, error) {
 	if deploy == nil || !shouldDelete(&deploy.ObjectMeta, &deploy.Spec.Template) {
 		return deploy, nil
 	}
 	return deploy, h.deployments.DeleteNamespaced(deploy.Namespace, deploy.Name, nil)
 }
 
+// OnDaemonSetChange checks if it should delete a namespaced daemon set and if so, deletes it.
 func (h *handler) OnDaemonSetChange(key string, ds *appsv1.DaemonSet) (runtime.Object, error) {
 	if ds == nil || !shouldDelete(&ds.ObjectMeta, &ds.Spec.Template) {
 		return ds, nil
 	}
 	return ds, h.daemonsetClient.DeleteNamespaced(ds.Namespace, ds.Name, nil)
 }
 
+// shouldDelete returns true if the CATTLE_SERVER: <SERVER_URL> env var is set on any container in the pod. This will
+// remove the deployment so Rancher can redeploy it. -- correct?
```

Comment:
Minor: Might want to say something along the lines of "will return true if the pod is a cluster-agent or node-agent pod on version v2.0-v2.4 in the cattle-system namespace".... I _think_ that's what it's doing?

Reply:
How version v2.0-v2.4?